plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -86,13 +91,35 @@ func (d *defaultServicePlan) Admit(a admission.Attributes) error {
 		glog.V(4).Info(msg)
 		return admission.NewForbidden(a, errors.New(msg))
 	}
-	if len(sc.Plans) > 1 {
+	// find all the service plans that belong to the service class
+
+	// TODO: Need to be careful here. Is it possible to have only one
+	// serviceplan available while others are still in progress?
+
+	// implement field selector
+
+	// loop over all service plans accumulate into slice
+	plans, err := d.spLister.List(labels.Everything())
+	// TODO filter `plans` down to only those owned by `sc`.
+
+	// check if there were any service plans
+	// TODO: in combination with not allowing classes with no plans, this should be impossible
+	if len(plans) <= 0 {
+		msg := fmt.Sprintf("no plans found at all")
+		glog.V(4).Info(msg)
+		return admission.NewForbidden(a, errors.New(msg))
+	}
+
+	// check if more than one service plan was specified and error
+	if len(plans) > 1 {
 		msg := fmt.Sprintf("ServiceClass %q has more than one plan, PlanName must be specified", instance.Spec.ServiceClassName)
 		glog.V(4).Info(msg)
 		return admission.NewForbidden(a, errors.New(msg))
 	}
+	// otherwise, by default, pick the only plan that exists for the service class
 
-	p := sc.Plans[0]
+	p := plans[0]
+	glog.Warningf("Using plan %+v", p)
```

Comment:
why the warning when we have info on the next line? seems like a debug thing.

Reply:
agree, I'm gonna go with debugging at the time.